pkg/local_object_storage/shard/gc.go

```diff
@@ -119,24 +98,11 @@ func (gc *gc) listenEvents() {
 			}
 
 			v.prevGroup.Wait()
-
-			v.prevGroup.Add(len(v.handlers))
-
-			for i := range v.handlers {
-				h := v.handlers[i]
-
-				err := gc.workerPool.Submit(func() {
-					h(event)
-					v.prevGroup.Done()
-				})
-				if err != nil {
-					gc.log.Warn("could not submit GC job to worker pool",
-						zap.Error(err),
-					)
-
-					v.prevGroup.Done()
-				}
-			}
+			v.prevGroup.Add(1)
+			go func() {
+				v.handler(event)
```

Comment:
moving from `v.handlers` slice to single `v.handler` is not actually related to the pool drop patch, it's an autonomous refactoring